tests/build_test.sh

```diff
@@ -53,8 +64,189 @@ tearDown()
   }
 }
 
+function test_parse_build_config_success_exit_code()
+{
+  build_config=()
+
+  parse_configuration "$KW_BUILD_CONFIG_SAMPLE" build_config
+  assertTrue 'kw failed to load a regular config file' "[ 0 -eq $? ]"
+}
+
+# Test if etc/kworkflow_template.config contains all the expected settings
+function test_parse_build_config_standard_config()
+{
+  build_config=()
+
+  declare -A expected_build_configurations=(
+    [arch]='x86_64'
+    [cpu_scaling_factor]='100'
+    [kernel_img_name]='bzImage'
+    [menu_config]='nconfig'
+    [doc_type]='htmldocs'
+  )
+
+  parse_configuration "$TMP_DIR/build.config" build_config
+  assertConfigurations build_config expected_build_configurations "$LINENO"
+
+  true # Reset return value
+}
+
+function assertConfigurations()
+{
+  declare -n configurations_ref=$1
+  declare -n expected_configurations_ref=$2
+  local lineno=${3:-LINENO}
+
+  # check if configurations is contained in expected_configurations
+  for k in "${!configurations_ref[@]}"; do
+    if [[ ${expected_configurations_ref[$k]+token} != token ]]; then
```

Comment:
`+token`? That's new for me. Could you elaborate on this?

Reply:
Actually, this is copy pasted, so I'm not sure about much of its workings, but I can try to improve it.

Reply:
I've done a quick research on this code, and could understand the following:
1. It was introduced by @matheustavares in 8e716c50, and
2. It aims at generalizing the process you use to verify those config arrays, the basic idea being that:
  1. You access array B using a key `$k` that was taken from array A.
  2. If B has such a key, then you end up with the string "token", otherwise it's empty (which is obviously different from "token" on the right hand side).
  3. Now you can check every key in the array easily.

Truly an ingenious solution!